### ISSUE TITLE
Provide a way to destroy authlink instance

### ISSUE THREAD

**Issue by ianhe8x** [subquery/network-clients#128]
There's some background tasks start with authlink instantiation. 

Ideally we should offer user a way to destroy them.

**Comment by stwiname** [comment]
This is causing issues with subql tests

```
Jest has detected the following 1 open handle potentially keeping Jest from exiting:

  ●  Timeout

      174 |
      175 |     if (this.nodeConfig.dictionaryResolver) {
    > 176 |       link = dictHttpLink({
          |                          ^
      177 |         authUrl: this.nodeConfig.dictionaryResolver,
      178 |         chainId,
      179 |         logger,

      at new AgreementManager (node_modules/@subql/apollo-links/src/agreementManager.ts:32:5)
      at deploymentHttpLink (node_modules/@subql/apollo-links/src/authHttpLink.ts:38:28)
      at dictHttpLink (node_modules/@subql/apollo-links/src/authHttpLink.ts:31:10)
      at new DictionaryService (packages/node-core/src/indexer/dictionary.service.ts:176:26)
      at Object.<anonymous> (packages/node-core/src/indexer/dictionary.service.test.ts:117:25)
```

**Comment by ianhe8x** [comment]
Fixed in v1.0.2 release
@stwiname You can update the test above.
